### ISSUE TITLE
Web setup miscounts the number of modules

### ISSUE THREAD

**Issue by brendanfalkowski** [magento/magento2#21700]
### Preconditions (*)

1. Magento 2.3.0 using a client's repo for initial setup

### Steps to reproduce (*)

1. Access sitename.test/setup
2. Proceed through the steps
3. In the step for module selection, the module count is wrong.

### Expected result (*)

The module count should be like "300 of 311".

### Actual result (*)

<img width="841" alt="Screen Shot 2019-03-11 at 3 50 15 PM" src="https://user-images.githubusercontent.com/214924/54163849-95392e80-4417-11e9-88a0-d7d82a583fe7.png">

Selecting or deselecting a module affects the left number, which is correct but the sum is incorrect.

I can reproduce this in a client repo, but it seems like this is a bug in the setup tool because none of that code is even executing at this point. It shouldn't matter what the contents of the module folders are when setup is run. The incorrect sum is possibly from counting modules wrong or the position of numbers is mixed up (and so is the check/uncheck evaluation).

**Comment by magento-engcom-team** [comment]
Hi @brendanfalkowski. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@brendanfalkowski do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @brendanfalkowski   GitHub is intended for Magento Open Source users to report on issues related to Open Source only.
You can report Commerce-related issues in one of two ways:
You can use the Support portal associated with your account
or
If you are a Partner reporting on behalf of a merchant, use the Partner portal.

**Comment by brendanfalkowski** [comment]
@magento-engcom-team — I am not a Partner, so I have no access to these tools, but I work almost exclusively on Magento Commerce (Enterprise) projects and have for years. 

There should be a clear path for people in my position to report bugs and improve the Commerce product (this is a goal I have) without paying for Partner status (this is not a goal I have).

**Comment by ghost** [comment]
@brendanfalkowski ok , but i'm not able to reproduce this on CE version. **GitHub is intended for Magento Open Source users**

**Comment by magento-engcom-team** [comment]
Hi @sdzhepa. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by sdzhepa** [comment]
Hello @brendanfalkowski 
Thank you for contribution and collaboration!

According to current Magento organization flows and processes issues related to Magento Commerce (Enterprise) can be reported only to Enterprise Support.

@brendanfalkowski  as I can see from your [comment](https://github.com/magento/magento2/issues/21700#issuecomment-472162129),  you are not a partner but you 

> work almost exclusively on Magento Commerce (Enterprise)

 I assume your employer or client has bought the license and should have the ability to report issue to Support.

I think the best approach will be to ask them to report this issue directly to Support.

From my side, I will try to search is this issue created in the internal Jira or not. If not I will create a new ticket but I can not promise anything. Priority and Severity for fix this issue is out of my responsibility.

**Comment by okorshenko** [comment]
Hi @brendanfalkowski 
We will double check this issue. But we need more details from your side. I believe this is not EE related issue and if the issue exists, it should be reproducible on CE as well. We will contact you shortly to get more information about this case. 

Thank you for follow up on this 👍

**Comment by sdzhepa** [comment]
Hello @brendanfalkowski 

We have investigated this issue on the latest Magento(Open Source and Commerce editions) based on the latest `2.3-develop` branch(es) 

For verifying we used next Preconditions and Scenarios


**Preconditions:**

1. Clone Magento from the git repository and do all required actions to be ready for web installation
2. In browser got to Web Setup: `sitename.test/setup`
3. On the "Welcome to Magento" page click *Agree and Setum Magento*
4. On "Readiness Check" do check and click *Next button*
5. On "Add a Database" page fill all required fields and click *Next*
6. On "Customize Your Store" click *Next*
 

**SCENARIO#1:** _**Count of modules on Web Page VS Summ number**_

1. On "Step 4: Customize Your Store" page
2. Click to open "Advanced Modules Configurations" section
3. Count all listed modules(check-boxes/Module labels) 
_Note_: I did it using XPath : `//div[@class='multiselect multiselect-custom']/ul/li/label`

***Actual Result = Expected Result***
- XPathes total amount = 230
- The sum of modules in the bottom of the page = 230
![i21700_sc1](https://user-images.githubusercontent.com/3409584/54453345-1ef14080-4725-11e9-8563-6480780ed700.jpg)


**SCENARIO#2:** **_Count of module folders on the filesystem VS. Sum and count on the web_**

1. Go to the filesystem with Magento and count number of folders in "app/code/Magento"
_Note_: I did it using next bash command : `/bin/ls -lA  | egrep -c '^d'`

***Actual Result = Expected Result***
- Count of the folders = 230
- The same as on Web Page
![i21700_sc2](https://user-images.githubusercontent.com/3409584/54453360-26184e80-4725-11e9-9a53-7b7b7069a819.jpg)



**SCENARIO#3:** Unselecting listed modules on the Web Install page
1. On "Step 4: Customize Your Store" step page
2. Click to open "Advanced Modules Configurations" section
3. One by one click to disable modules and check rest total after each click

_Link to the GIF video of performed steps:_ https://www.screencast.com/t/mC2Ewp3pG12

***Actual Result = Expected Result***
- After each click(disable module) Total summ still = 230
- After each click(disable module) Count of still enabled modules reflects user click




**Summarize:** 
Unfortunately, we are not able to reproduce the issue by steps from the description.
It seems some important precondition(s) or step(s) was missed.

@brendanfalkowski , Could you please help us and provide missing information?
- Which type of install did you use?
- Are you able to reproduce this issue on the clear Magento installation? I mean without any additional customization and extensions just based on lates `2.3-develop` branch

**Comment by Mobecls** [comment]
Hey @brendanfalkowski, closing this issue as it looks like we were not able to reproduce this issue. 

Please, see the comment https://github.com/magento/magento2/issues/21700#issuecomment-473393787

Closing because of inactivity.
